### ISSUE TITLE
Failed saving the best model according to hedwig/models/bert/README.md

### ISSUE THREAD

**Issue by fengyunyi** [castorini/hedwig#31]
Tried the steps according to hedwig/models/bert/README.md but the best model did not get saved. New to pytorch, which part of the code does this job?

**Comment by achyudh** [comment]
Please check lines 121 to 125 here: https://github.com/castorini/hedwig/blob/master/common/trainers/bert_trainer.py

**Comment by fengyunyi** [comment]
Got it. Changed the path name on the README.md and it will work.

**Comment by amirmohammadkz** [comment]
> 
> 
> Got it. Changed the path name on the README.md and it will work.

Can you explain which part exactly should I change? After finetuning I have only a checkpoint file in checkpoint folder which its name timestamp refers to the end of the first epoch time

**Comment by fengyunyi** [comment]
> > Got it. Changed the path name on the README.md and it will work.
> 
> Can you explain which part exactly should I change? After finetuning I have only a checkpoint file in checkpoint folder which its name timestamp refers to the end of the first epoch time

I think that is it.

**Comment by amirmohammadkz** [comment]
> 
> 
> > > Got it. Changed the path name on the README.md and it will work.
> > 
> > 
> > Can you explain which part exactly should I change? After finetuning I have only a checkpoint file in checkpoint folder which its name timestamp refers to the end of the first epoch time
> 
> I think that is it.

Thanks! But another question. In the finetuning process, I think for every epoch we get accuracy for dev set and at the end of finetuning, The model picks the best dev acc and tests it on the test dataset( so we can see the best result at the end of the output).
but I was wondering when I am running the test process on the checkpoint(I mentioned before), sometimes the dev acc that is shown is not included in the finetuning process outputs and often it is lower than the best dev acc. Let me show an example:

output of finetuning(I skip unnesseray logs):

```
Device: CUDA
Number of GPUs: 1
Distributed training: False
FP16: False
100% 231508/231508 [00:00<00:00, 430268.33B/s]
100% 407873900/407873900 [00:32<00:00, 12631201.67B/s]
Number of examples:  5661
Batch size: 16
Num of steps: 7060

Epoch Iteration Progress   Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
    1       354      1/20    0.5730   0.5730   0.5730   0.5730     0.6836
Epoch Iteration Progress   Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
    2       708      2/20    0.5842   0.5842   0.5842   0.5842     0.6809
Epoch Iteration Progress   Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
    3      1062      3/20    0.5842   0.5842   0.5842   0.5842     0.6810
Epoch Iteration Progress   Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
    4      1416      4/20    0.5817   0.5817   0.5817   0.5817     0.9512
Epoch Iteration Progress   Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
    5      1770      5/20    0.5755   0.5755   0.5755   0.5755     1.4543
Epoch Iteration Progress   Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
    6      2124      6/20    0.6064   0.6064   0.6064   0.6064     2.0489
Epoch Iteration Progress   Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
    7      2478      7/20    0.5854   0.5854   0.5854   0.5854     2.1437
Epoch Iteration Progress   Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
    8      2832      8/20    0.5990   0.5990   0.5990   0.5990     2.2655
Epoch Iteration Progress   Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
    9      3186      9/20    0.5817   0.5817   0.5817   0.5817     2.2572
Epoch Iteration Progress   Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
   10      3540     10/20    0.5965   0.5965   0.5965   0.5965     2.3332
Epoch Iteration Progress   Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
   11      3894     11/20    0.5854   0.5854   0.5854   0.5854     2.4956
Early Stopping. Epoch: 10, Best Dev F1: 0.6064356435643564
Epoch:  50% 10/20 [1:50:36<1:40:21, 602.18s/it]Inference time 29.027603149414062

Split  Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
  DEV    0.6064   0.6064   0.6064   0.6064     2.0489
Inference time 25.89557933807373

Split  Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
 TEST    0.6092   0.6092   0.6092   0.6092     2.0600

```
output of testing:

```
Device: CUDA
Number of GPUs: 1
Distributed training: False
FP16: False
Inference time 8.667673826217651

Split  Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
  DEV    0.5199   0.5199   0.5199   0.5199     0.7643
Inference time 7.662160158157349

Split  Dev/Acc.  Dev/Pr.  Dev/Re.   Dev/F1   Dev/Loss
 TEST    0.6341   0.6341   0.6341   0.6341     0.6951
```
As you see, the dev acc of testing output is lower than the best dev acc of finetuning output but I can not find the accs of testing output in finetuning output. why?

**Comment by fengyunyi** [comment]
I don't know. I tried but it was the same. The code looks same too.
